cms/djangoapps/contentstore/views/course.py

```diff
@@ -1907,3 +1911,34 @@ def _get_course_creator_status(user):
         course_creator_status = 'granted'
 
     return course_creator_status
+
+
+def get_allowed_organizations(user):
+    """
+    Helper method for returning the list of organizations for which the user is allowed to create courses.
+    """
+    if settings.FEATURES.get('ENABLE_CREATOR_GROUP', False):
+        return get_organizations(user)
+    else:
+        return []
+
+
+def user_can_create_organizations(user):
+    """
+    Returns True if the user can create organizations.
+    """
+    return user.is_staff or not settings.FEATURES.get('ENABLE_CREATOR_GROUP', False)
+
+
+def get_organizations(user):
+    """
+    Returns the list of organizations for which the user is allowed to create courses.
+    """
+    user = CourseCreator.objects.filter(user=user)
+    if user.count() == 0:
+        return []
+    elif user[0].all_organizations:
+        # User is defined to be unique, can assume a single entry.
+        return Organization.objects.all().values_list('short_name', flat=True)
+    else:
+        return user[0].organizations.all().values_list('short_name', flat=True)
```

Comment:
Also, I believe some testing would be beneficial. You could add tests to `test_organizations`, extend the existing ones or create a new test suite.

Reply:
what I see is that you're only testing get_allowed_organizations

Reply:
More test have been added to can_create_courses. This permission is only validation on the frontend, because this permission is checked once a request is made to create a course or library in the backend